src/providers/WorkflowCore.QueueProviders.SqlServer/Services/SqlServerQueueProvider.cs

```diff
@@ -0,0 +1,157 @@
+        private readonly ISqlServerQueueProviderMigrator _migrator;
+        private readonly ISqlCommandExecutor _sqlCommandExecutor;
+
+        private readonly string _queueWork;
+        private readonly string _dequeueWork;
+
+        public SqlServerQueueProvider(IServiceProvider serviceProvider, SqlServerQueueProviderOption opt)
+        {
+            _connectionString = opt.ConnectionString;
+            _canMigrateDb = opt.CanMigrateDb;
+            _canCreateDb = opt.CanCreateDb;
+
+            _names = serviceProvider.GetService<IBrokerNamesProvider>()
+                     ?? new BrokerNamesProvider(opt.WorkflowHostName);
+            _sqlCommandExecutor = serviceProvider.GetService<ISqlCommandExecutor>()
+                                  ?? new SqlCommandExecutor();
+            _migrator = serviceProvider.GetService<ISqlServerQueueProviderMigrator>()
+                        ?? new SqlServerQueueProviderMigrator(opt.ConnectionString, _names, _sqlCommandExecutor);
+
+            IsDequeueBlocking = true;
+
+            _queueWork = GetFromResource("QueueWork");
+            _dequeueWork = GetFromResource("DequeueWork");
+        }
+
+        private static string GetFromResource(string file)
+        {
+            var resName = $"WorkflowCore.QueueProviders.SqlServer.Services.{file}.sql";
+
+            using (var reader = new StreamReader(Assembly.GetExecutingAssembly().GetManifestResourceStream(resName)))
+            {
+                return reader.ReadToEnd();
+            }
+        }
+
+
+        public bool IsDequeueBlocking { get; }
+
+        public async Task Start()
+        {
+            if (_canCreateDb) _migrator.CreateDb();
+            if (_canMigrateDb) _migrator.MigrateDb();
+        }
+
+        public async Task Stop()
+        {
+            // Do nothing
+        }
+
+        public void Dispose()
+        {
+            Stop().Wait();
+        }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Write a new id to the specified queue
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="queue"></param>
+        /// <returns></returns>
+        public async Task QueueWork(string id, QueueType queue)
+        {
+            if (String.IsNullOrEmpty(id)) throw new ArgumentNullException(nameof(id), "Param id must not be null");
+
+            SqlConnection cn = null;
+            try
+            {
+                string msgType, initiatorService, targetService, contractName;
+                if (queue == QueueType.Workflow)
+                {
+                    msgType = _names.WorkflowMessageType;
+                    initiatorService = _names.InitiatorWorkflowServiceName;
+                    targetService = _names.TargetWorkflowServiceName;
+                    contractName = _names.WorkflowContractName;
+                } else
+                {
+                    msgType = _names.EventMessageType;
```

Comment:
I think it may be cleaner to have methods on the name provider service that pass in a queue type and determine the various object names in there instead of in this class?